build/Dockerfile

```diff
@@ -33,4 +35,4 @@ ARG git_sha=NONE
 LABEL multi.GIT_URL=${git_url} \
       multi.GIT_SHA=${git_sha} \
       app=hyperconverged-cluster-operator \
-      arch=${ARCH}
+      golang.build.platform=${TARGETPLATFORM}
```

Comment:
`arch` label is not needed for HCO images?

Reply:
this is a label I added in the previous PR, for us, for debug. not a requirement. just rename it. We also get the `architecture` label from the base images.